test/integration/gnmiutils.go

```diff
@@ -74,50 +75,50 @@ func GNMIGet(ctx context.Context, c client.Impl, paths []DevicePath) ([]DevicePa
 	getTZRequest := &gpb.GetRequest{}
 	if err := proto.UnmarshalText(protoString, getTZRequest); err != nil {
 		fmt.Printf("unable to parse gnmi.GetRequest from %q : %v", protoString, err)
-		return nil, err
+		return nil, nil, err
 	}
 
 	response, err := c.(*gclient.Client).Get(ctx, getTZRequest)
 	if err != nil || response == nil {
-		return nil, err
+		return nil, nil, err
 	}
-
-	return convertGetResults(response)
+	pathsResp, errResp := convertGetResults(response)
+	return pathsResp, getTZRequest.Extension, errResp
 }
 
 // GNMISet generates a SET request on the given client for a path on a device
-func GNMISet(ctx context.Context, c client.Impl, devicePaths []DevicePath) (string, error) {
+func GNMISet(ctx context.Context, c client.Impl, devicePaths []DevicePath) (string, []*gnmi_ext.Extension, error) {
 	var protoBuilder strings.Builder
 	for _, devicePath := range devicePaths {
 		protoBuilder.WriteString(MakeProtoUpdatePath(devicePath))
 	}
 
 	setTZRequest := &gpb.SetRequest{}
 	if err := proto.UnmarshalText(protoBuilder.String(), setTZRequest); err != nil {
-		return "", err
+		return "", nil, err
 	}
 
 	setResult, setError := c.(*gclient.Client).Set(ctx, setTZRequest)
 	if setError != nil {
-		return "", setError
+		return "", nil, setError
 	}
-	return extractSetTransactionID(setResult), nil
+	return extractSetTransactionID(setResult), setResult.Extension, nil
 }
 
 // GNMIDelete generates a SET request on the given client to delete a path for a device
-func GNMIDelete(ctx context.Context, c client.Impl, devicePaths []DevicePath) error {
+func GNMIDelete(ctx context.Context, c client.Impl, devicePaths []DevicePath) ([]*gnmi_ext.Extension, error) {
 	var protoBuilder strings.Builder
 	for _, devicePath := range devicePaths {
 		protoBuilder.WriteString(MakeProtoDeletePath(devicePath.deviceName, devicePath.path))
 	}
 
 	setTZRequest := &gpb.SetRequest{}
 	if err := proto.UnmarshalText(protoBuilder.String(), setTZRequest); err != nil {
-		return err
+		return nil, err
 	}
 
 	_, err := c.(*gclient.Client).Set(ctx, setTZRequest)
-	return err
+	return setTZRequest.Extension, err
```

Comment:
same here - should you be getting the response and extracting the Extension from it?

Reply:
same